public/app/features/explore/utils/links.test.ts

```diff
@@ -145,6 +145,52 @@ describe('getFieldLinksForExplore', () => {
     });
     const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, range, dataFrame });
     expect(links).toHaveLength(1);
+    expect(links[0].href).toBe(
+      '/explore?left=%7B%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%2C%22datasource%22%3A%22uid_1%22%2C%22queries%22%3A%5B%7B%22query%22%3A%22query_1-foo%22%7D%5D%2C%22panelsState%22%3A%7B%7D%7D'
+    );
+  });
+
+  it('returns internal links when target contains targetField template variable', () => {
+    const { field, range, dataFrame } = setup({
+      title: '',
+      url: '',
+      internal: {
+        query: { query: 'query_1-${__targetField}' },
+        datasourceUid: 'uid_1',
+        datasourceName: 'test_ds',
+      },
+    });
+    const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, range, dataFrame });
+    expect(links).toHaveLength(1);
+    expect(links[0].href).toBe(
+      '/explore?left=%7B%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%2C%22datasource%22%3A%22uid_1%22%2C%22queries%22%3A%5B%7B%22query%22%3A%22query_1-foo%22%7D%5D%2C%22panelsState%22%3A%7B%7D%7D'
+    );
+  });
+
+  it('returns internal links when target contains field name template variable', () => {
+    // field cannot be hyphenated, change field name to non-hyphenated
+    const noHyphenLink = {
+      title: '',
+      url: '',
+      internal: {
+        query: { query: 'query_1-${fluxDimensions}' },
+        datasourceUid: 'uid_1',
+        datasourceName: 'test_ds',
+      },
+    };
+    const { field, range, dataFrame } = setup(noHyphenLink, true, {
+      name: 'fluxDimensions',
+      type: FieldType.string,
+      values: new ArrayVector([ROW_WITH_TEXT_VALUE.value, ROW_WITH_NULL_VALUE.value]),
+      config: {
+        links: [noHyphenLink],
+      },
+    });
+    const links = getFieldLinksForExplore({ field, rowIndex: ROW_WITH_TEXT_VALUE.index, range, dataFrame });
+    expect(links).toHaveLength(1);
+    expect(links[0].href).toBe(
+      '/explore?left=%7B%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%2C%22datasource%22%3A%22uid_1%22%2C%22queries%22%3A%5B%7B%22query%22%3A%22query_1-foo%22%7D%5D%2C%22panelsState%22%3A%7B%7D%7D'
```

Comment:
This is hard to read. Could we mock `generateInternalHref` so it returns only the interpolated query?

Reply:
Unfortunately that is a private function inside `dataLinks` - I could mock `mapInternalLinkToExplore` but I really like the explicit testing that the interpolation happens as expected. I changed this where the URL is not encoded in the test file, that's what was happening in another place in this test. What do you think?